graylog2-server/src/main/java/org/graylog2/migrations/V20190705071400_AddEventIndexSetsMigration.java

```diff
@@ -96,32 +110,35 @@ public void upgrade() {
                 "Stores Graylog system events.",
                 elasticsearchConfiguration.getDefaultSystemEventsIndexPrefix(),
                 ElasticsearchConfiguration.DEFAULT_SYSTEM_EVENTS_INDEX_PREFIX,
-                Stream.DEFAULT_SYSTEM_EVENTS_STREAM_ID,
+                SYSTEM_EVENT_TEMPLATE_TYPE,
+                DEFAULT_SYSTEM_EVENTS_STREAM_ID,
                 "All system events",
                 "Stream containing all system events created by Graylog"
         );
+        ensureSystemNotificationEventsDefinition();
     }
 
     private void ensureEventsStreamAndIndexSet(String indexSetTitle,
                                                String indexSetDescription,
                                                String indexPrefix,
                                                String indexPrefixConfigKey,
+                                               String indexTemplate,
                                                String streamId,
                                                String streamTitle,
                                                String streamDescription) {
-        checkIndexPrefixConflicts(indexPrefix, indexPrefixConfigKey);
+        checkIndexPrefixConflicts(indexPrefix, indexPrefixConfigKey, indexTemplate);
 
-        final IndexSet eventsIndexSet = setupEventsIndexSet(indexSetTitle, indexSetDescription, indexPrefix);
+        final IndexSet eventsIndexSet = setupEventsIndexSet(indexSetTitle, indexSetDescription, indexPrefix, indexTemplate);
         try {
             streamService.load(streamId);
         } catch (NotFoundException ignored) {
             createEventsStream(streamId, streamTitle, streamDescription, eventsIndexSet);
         }
     }
 
-    private void checkIndexPrefixConflicts(String indexPrefix, String configKey) {
+    private void checkIndexPrefixConflicts(String indexPrefix, String configKey, String templateType) {
         final DBQuery.Query query = DBQuery.and(
-                DBQuery.notEquals(IndexSetConfig.FIELD_INDEX_TEMPLATE_TYPE, Optional.of(EVENT_TEMPLATE_TYPE)),
+                DBQuery.notEquals(IndexSetConfig.FIELD_INDEX_TEMPLATE_TYPE, Optional.of(templateType)),
```

Comment:
This only works for new installations. on upgrades, this will fail because the system event index already exists with `template_type` `events`

Reply:
I now check for the upgrade case and delete the indexset (plus associated ranges and field types) and stream if necessary. Then we can create a clean new index set and stream regardless.